webui-src/app/channels/channels.js

```diff
@@ -1,25 +1,69 @@
 const m = require('mithril');
+const widget = require('widgets');
+const rs = require('rswebui');
+const util = require('channels/channels_util');
 
-const Channel = () => {
-  return {
-    view: () => m('.channel'),
-  };
+const getChannels = {
+  All: [],
+  PopularChannels: [],
+  SubscribedChannels: [],
+  async load() {
+    await rs.rsJsonApiRequest('/rsgxschannels/getChannelsSummaries', {}, (data) => {
```

Comment:
The method you are using is correct, but it has not much effect. The async method was intended to be directly used with the resolved promise.
So you can simply get the data from the await, like:
```js
const data = await rs.rsJsonApiRequest('/rsgxschannels/getChannelsSummaries');
```
And you won't have to pass a callback for using the data.

Reply:
Okay, this is a lot more concise.
Also I am using .then() for further use of the data. I don't know if it is really necessary.